### PR TITLE
Activity log: 'restore' action types and actions

### PR DESCRIPTION
**Pull request by seear**
Another chunk of #10862 to merge. This adds the actions and http handling for restoring a backup.

No visual changes.

**To Test**
* Includes unit tests
* Network request can be fired with following command in the console:
`dispatch( { type: 'REWIND_RESTORE', siteId: '{site}', timestamp: {timestamp} } )`
(timestamp is unix timestamp like `1496769219`)

### CODE REVIEW

File: client/state/data-layer/wpcom/activity-log/rewind/to/index.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+const requestRestore = ( { dispatch }, action ) => {
+	dispatch( http( {
+		method: 'POST',
+		path: `/activity-log/${ action.siteId }/rewind/to/${ action.timestamp }`,
+		apiVersion: '1',
+	}, action ) );
+};
+
+export const receiveRestoreSuccess = ( { dispatch }, { siteId, timestamp } ) => {
+	dispatch( rewindCompleteRestore( siteId, timestamp ) );
```
Comment (seear) [inline client/state/data-layer/wpcom/activity-log/rewind/to/index.js]: I think this API call will actually return immediately, so it is wrong to say that the restore is complete at this point. However, until we add code to poll D5890-code, we can probably leave this here?

Comment (sirreal) [inline client/state/data-layer/wpcom/activity-log/rewind/to/index.js]: 👍  sounds good as far as this PR is concerned.

**Comment by seear** [comment]
> I'm not clear how I can test successful requests.

Right, me neither. I took the unit test data from the API definition in wpcom.

File: client/state/data-layer/wpcom/activity-log/rewind/to/index.js (hunk 1 above)
Comment (dereksmart) [inline client/state/data-layer/wpcom/activity-log/rewind/to/index.js]: +1, this is good for now!

**Comment by dereksmart** [comment]
LGTM!

**Pull request merged**